pkg/osquery/enrollment_details.go

```diff
@@ -31,13 +37,16 @@ func getEnrollDetails(ctx context.Context, osquerydPath string) (service.Enrollm
 		return details, errors.New("Skipping enrollment details")
 	}
 
+	details.OSPlatform = runtime.GOOS
+	details.OSPlatformLike = runtime.GOOS
```

Comment:
Anything we can do here, we should be pulling out of the osquery query

Reply:
I just started with these 2 because I know they are consistent with osq. However, hostname is not consistent. On my mac, runtime pkg give a different value that osquery for hostname. Do you think we should just avoid inconsistent values all together? or grab what we can using go, then let osq overwrite the values.

Reply:
I think these two are the most important. So if we don't have an answer for the others, even just having these is good.

macOS has 3 different hostnames. There might be a way to get the others

Reply:
I think we would have to exec to get the other hostnames. I suppose in theory we could replicate what osquery does on all platforms with execs, but I'm not sure if we want to go down that path. Or we can just do it for data that is critical to enrollment.